zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -63,85 +61,52 @@ pub(super) const INDEX_ACCOUNTS_UIVK: &str =
 pub(super) const INDEX_HD_ACCOUNT: &str =
     r#"CREATE UNIQUE INDEX hd_account ON accounts (hd_seed_fingerprint, hd_account_index)"#;
 
-/// Stores diversified Unified Addresses that have been generated from accounts in the
-/// wallet.
+/// Stores addresses that have been generated from accounts in the wallet.
+///
+/// ### Columns
 ///
-/// - The `cached_transparent_receiver_address` column contains the transparent receiver component
-///   of the UA. It is cached directly in the table to make account lookups for transparent outputs
-///   more efficient, enabling joins to [`TABLE_TRANSPARENT_RECEIVED_OUTPUTS`].
+/// - `account_id`: the account whose IVK was used to derive this address.
+/// - `diversifier_index_be`: the diversifier index at which this address was derived.
+/// - `key_scope`: the key scope for which this address was derived.
+/// - `address`: The Unified, Sapling, or transparent address. For Unified and Sapling addresses,
+///   only external-key scoped addresses should be stored in this table; for purely transparent
+///   addresses, this may be an internal-scope (change) address, so that we can provide
+///   compatibility with HD-derived change addresses produced by transparent-only wallets.
+/// - `transparent_child_index`: the diversifier index, if it is in the range of a non-hardened
+///   transparent address index. This is used for gap limit handling and is always populated if the
+///   diversifier index is in that range; since the diversifier index is stored as a byte array we
+///   cannot use SQL integer operations on it and thus need it as an integer as well.
+/// - `cached_transparent_receiver_address`: the transparent receiver component of address (which
+///   may be the same as `address` in the case of an internal-scope transparent change address or a
+///   ZIP 320 interstitial address). It is cached directly in the table to make account lookups for
+///   transparent outputs more efficient, enabling joins to [`TABLE_TRANSPARENT_RECEIVED_OUTPUTS`].
```

Comment:
This documentation is now AFAICT incorrect. It is not "the transparent receiver component of address" because there are cases where `address` does not contain a transparent receiver, and yet we still populate this column.

Assuming that this is guaranteed to always be the transparent receiver derived from the same diversifier index as `address`:
```suggestion
/// - `cached_transparent_receiver_address`: the transparent address derived from the same
///   viewing key and at the same diversifier index as `address`. This may be the same as
///   `address` in the case of an internal-scope transparent change address or a
///   ZIP 320 interstitial address, and it may be a receiver within `address` in the case of a
///   Unified Address with transparent receiver. It is cached directly in the table to make account lookups for
///   transparent outputs more efficient, enabling joins to [`TABLE_TRANSPARENT_RECEIVED_OUTPUTS`].
```